tests/api_tests/nodes/test_views.py

```diff
@@ -3079,7 +3095,14 @@ def setUp(self):
         self.public_pointer = self.public_project.add_pointer(self.public_pointer_project,
                                                               auth=Auth(self.user),
                                                               save=True)
-        self.public_url = '/{}nodes/{}/node_links/{}'.format(API_BASE, self.public_project._id, self.public_pointer._id)
+        self.public_url = '/{}nodes/{}/node_links/{}/'.format(API_BASE, self.public_project._id, self.public_pointer._id)
+
+    def test_delete_node_link_no_permissions_for_target_node(self):
+        pointer_project = ProjectFactory(creator=self.user_two, is_public=False)
+        pointer = self.public_project.add_pointer(pointer_project, auth=Auth(self.user), save=True)
+        url = '/{}nodes/{}/node_links/{}/'.format(API_BASE, self.public_project._id, pointer._id)
+        res = self.app.delete_json_api(url, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 204)
```

Comment:
In addition to checking status code, you can optionally reload the project and check that the actual expected result (removal of link) has occurred.